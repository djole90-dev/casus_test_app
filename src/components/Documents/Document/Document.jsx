import React from 'react';
import { DocumentBody, Title } from './Document.styles';
import C from '../C_Logo/C_Logo'

const Document = ({ title, category }) => (
  <DocumentBody>
    <Title>{title}</Title>
    <C category={category}/>
  </DocumentBody>
);

export default Document;
