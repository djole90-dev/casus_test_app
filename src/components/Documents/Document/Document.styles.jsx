import styled from 'styled-components';
import { Box, Typography } from '@material-ui/core';

export const DocumentBody = styled(Box)`
  width:192px;
  height: 240px;
  padding: 32px 12px 12px 24px;
  display:flex;
  flex-direction: column;
  background: #ffffff;
  margin-bottom:39px;
  margin-left:32px;
  cursor: pointer;
  position:relative;
`;
export const Title = styled(Typography)`
  font-size:15px;
  line-height: 1.33;
  letter-spacing: 0.21px;
  color: ${({ theme }) => theme.palette.primary.main};
  font-weight:500;
`;

export const C = styled('img')`
  height:18px;
  width: 16px;
  position: absolute;
  left: 14px;
  bottom: 12px;
`;
