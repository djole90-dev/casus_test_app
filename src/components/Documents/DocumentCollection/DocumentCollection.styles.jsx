import styled from 'styled-components';
import { Box } from '@material-ui/core';

export const Container = styled(Box)`
  width: 100%;
  display:flex;
  flex-wrap: wrap;
`;
