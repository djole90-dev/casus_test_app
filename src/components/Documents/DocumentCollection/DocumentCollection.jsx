import React from 'react';
import { Container } from './DocumentCollection.styles';
import Document from '../Document/Document';

const DocumentCollection = ({ documents }) => {
  return (
    <Container>
      {documents.map((doc, i) => <Document key={i} title={doc.title} category={doc.category} />)}
    </Container>
  );
};

export default DocumentCollection;
