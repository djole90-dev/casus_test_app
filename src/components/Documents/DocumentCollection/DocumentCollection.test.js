import React from 'react';
import { shallow } from 'enzyme';
import DocumentCollection from './DocumentCollection';
import Document from '../Document/Document';
import { Container } from './DocumentCollection.styles';

const setUp = (Component, props = {}) => {
  const component = shallow(<Component {...props} />);
  return component;
};

describe('Document Collection Component', () => {
  const baseProps = {
    documents: [
      { title: '123', category: '123' },
      { title: '123', category: '1234' },
      { title: '123', category: '5' }
    ]
  };

  it('should match snapshot', () => {
    const component = setUp(DocumentCollection, baseProps);
    expect(component).toMatchSnapshot();
  });

  it('should render properly with documents', () => {
    const component = setUp(DocumentCollection, baseProps);
    const container = component.find(Container)
    const docItems = container.children().find(Document)

    expect(container.length).toBe(1);
    expect(docItems.length).toBe(3);
  });

  it('should render properly without documents', () => {
    const component = setUp(DocumentCollection, { documents: []});
    const container = component.find(Container)
    const docItems = container.children().find(Document)

    expect(container.length).toBe(1);
    expect(docItems.length).toBe(0);
  });
});
