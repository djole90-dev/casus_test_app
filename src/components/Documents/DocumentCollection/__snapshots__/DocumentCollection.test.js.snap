// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Document Collection Component should match snapshot 1`] = `
<Styled(Styled(MuiBox))>
  <Document
    category="123"
    key="0"
    title="123"
  />
  <Document
    category="1234"
    key="1"
    title="123"
  />
  <Document
    category="5"
    key="2"
    title="123"
  />
</Styled(Styled(MuiBox))>
`;
